tests/test_SequenceManager.cc

```diff
@@ -529,3 +533,266 @@ TEST_CASE("SequenceManager: store_sequence() & load_sequence() - Empty sequence"
         REQUIRE(seq_deserialized.empty());
     }
 }
+
+
+TEST_CASE("SequenceManager: git repository", "[SequenceManager]")
+{
+
+    SECTION("Create and store sequence")
+    {
+        // clean surrounding
+        std::filesystem::remove_all(temp_dir);
+
+        // init manager
+        SequenceManager manager{ temp_dir};
+
+        // create sequence
+        Sequence seq{ "git Sequence 1",  SequenceName{ "git_sequence_1" }, UniqueId{ 0xabcdef123456 } };
+        Step step_01{Step::type_while};
+        step_01.set_label("while");
+        step_01.set_script("return i < 10");
+        seq.push_back(step_01);
+
+        // store sequence and make git commit
+        manager.store_sequence(seq);
+
+        // unable to reach gl debug functions without a new function in sequenceManager
+        // Therefore a standalone git handler has to be initialized
+        git::GitRepository gl{temp_dir};
+
+        // check if commit was done
+        const std::string& expected_msg = "change sequence:\n- Add sequence.lua from new sequence 'git_sequence_1[0000abcdef123456]'\n- Add step_1_while.lua from new sequence 'git_sequence_1[0000abcdef123456]'";
+        const auto& last_msg = gl.get_last_commit_message();
+        REQUIRE(expected_msg == last_msg);
+
+
+        // use raw repository to access status
+        const auto stats = gl.status();
+        REQUIRE(stats.size() != 0);
+        bool seq_exists = false;
+        for(const auto& elm: stats)
+        {
+            if (gul14::starts_with(elm.path_name, "git_sequence_1"))
+            {
+                seq_exists = true;
+
+                // if staging did not work: 
+                //      elm.handling == "untracked"
+                //      elm.changes == "untracked"
+                // if commit did not work
+                //      elm.handling == "staged"
+                //      elm.changes == "new file"
+                REQUIRE(elm.handling == "unchanged");
+                REQUIRE(elm.changes == "unchanged");
+
+            }
+        }
+        // check if created sequence is at least indexed by git
+        REQUIRE(seq_exists);
+    }
+
+    SECTION("change and store sequence (step_setup)")
+    {
+        SequenceManager manager{ temp_dir};
+
+        // load sequence an change step setup
+        auto seq = manager.load_sequence(UniqueId{ 0xabcdef123456 });
+        const auto step_setup_script =
+        R"(
+        a = 'Bob'
+        function test(name)
+            return name .. ' with some funky stuff!'
+        end
+        b = test('Alice') \t\b\b  c = 4)";
+        seq.set_step_setup_script(step_setup_script);
+
+        // store sequence
+        manager.store_sequence(seq);
+
+        git::GitRepository gl{temp_dir};
+
+        const std::string& expected_msg = "change sequence:\n- modify 'git_sequence_1[0000abcdef123456]/sequence.lua'";
+        const auto& last_msg = gl.get_last_commit_message();
```

Comment:
Again, why references? `const std::string` is simpler, clearer, and much safer.

Maybe as a rule of thumb: References (especially `const T&`) are great as function parameters when we want to avoid copying the given argument. For storing things for later use (like here), they are potentially dangerous because we never know if the thing they point to might be changed at some point.